### DISCUSSION TITLE
Can we create Package and Function in different namespaces?

### DISCUSSION THREAD

**Discussion by saltbo** [fission/fission#2676] [Q&A] [answered]
Can we create Package and Function in different namespaces, and a function can ref a Package from the other namespace?

If we can, the checks from `preupgradechecks`  should be update?
https://github.com/fission/fission/blob/main/cmd/preupgradechecks/checks.go#L149

If we can't. why not deny it when the Function is created?

**Comment by shubham-bansal96** [comment, marked as answer]
From version 1.18, fission don't allow creating resources in cross namespace.

Recently we have introduced validation controller, here we can think of adding a check to deny creating a function if package is in another namespace.
